src/steps/set-error-map.js

```diff
@@ -0,0 +1,43 @@
+import { CODES } from "../utils/report";
+
+export function setErrorMap({ reporter }) {
+  if (reporter.setErrorMap) {
+    reporter.setErrorMap({
+      [CODES.WordPressFilters]: {
+        text: (context) => context.message,
+        level: "ERROR",
+        category: "USER",
+      },
+      [CODES.BadUrl]: {
+        text: (context) => context.message,
+        level: "ERROR",
+        category: "USER",
+      },
+      [CODES.CustomUserCode]: {
+        text: (context) => context.message,
+        level: "ERROR",
+        category: "USER",
+      },
+      [CODES.Permissions]: {
+        text: (context) => context.message,
+        level: "ERROR",
+        category: "USER",
+      },
+      [CODES.Authentication]: {
+        text: (context) => context.message,
+        level: "ERROR",
+        category: "USER",
+      },
+      [CODES.Timeout]: {
+        text: (context) => context.message,
+        level: "ERROR",
+        category: "USER",
+      },
+      [CODES.UnknownError]: {
```

Comment:
Should we have this? For now we are treating errors without a code as "unknown". I would lean towards removing this and trying to have more precise errors.

However in core we have a example that kind of supports this usage. Error id `85901` seems to be a generic graphql error. It tells us that the error is graphql related, but not much more. Similarly this error could mean "unknown/generic error in Wordpress plugin", although I'd prefer to solve that by making sure we include the plugin name in structured logs.

Reply:
I am happy to go either way on that one. If nobody has has a differing opinion I have no problem removing this one.

Reply:
Speaking of 85901... is there any logic for how these codes are chosen? Guidelines for coming up with good codes? Trying to come up with a scheme that will allow me to create codes that are unique, meaningful, and easy to understand. Drawing a blank so far.

Reply:
My understanding is that the first 3 digits are the group (arbitrarily chosen) and then we count inside that group. I read that somewhere, but I can't find it now. I think @LekoArts knows!

Reply:
Here's an explanation (also see following message):

https://github.com/gatsbyjs/gatsby/pull/27176#issuecomment-703611482

Reply:
Thank you, you're the best ❤️

Reply:
Excellent, thank you for this. The ability to search for the code is one advantage I hadn't considered so I may rethink these codes a bit based on that.